testcafe/tests/photos_crud.js

```diff
@@ -0,0 +1,255 @@
+const _ = require("lodash");
+
+import Page from "./pages/photos-model";
+
+const page = new Page();
+
+fixture`PHOTOS - CRUD`.page`${config.photosUrl}`.beforeEach(async t => {
+  await t.useRole(regularUser);
+  await t.resizeWindow(1280, 1024); // No upload button on mobile view, beware of the size!
+
+  t.ctx.allPhotosStartCount = await page.allPhotos.count; //Pics count at test start
+  console.log("beforeEach > allPhotosStartCount " + t.ctx.allPhotosStartCount);
+});
+
+test("Uploading 1 pic from Photos view", async t => {
+  //new pic shows up
+
+  await t
+    .setFilesToUpload(page.btnUpload, ["../data/IMG0.jpg"])
+    .expect(page.divUpload.visible)
+    .ok()
+    .expect(page.modalUpload.exists)
+    .ok({ timeout: 50000 })
+    .expect(page.divUpload.child("h4").innerText)
+    .contains("Uploaded 1 out of 1 successfully");
+  await t.takeScreenshot();
+
+  const allPhotosEndCount = await page.allPhotos.count; //Pics count at the end
+  console.log("allPhotosEndCount " + allPhotosEndCount);
+  await t.expect(allPhotosEndCount).eql(t.ctx.allPhotosStartCount + 1);
+});
+
+test("Uploading 3 pcis from Photos view", async t => {
+  //new pics show up
+
+  await t
+    .setFilesToUpload(page.btnUpload, [
+      "../data/IMG-JPG.jpg",
+      "../data/IMG-PNG.png",
+      "../data/IMG-GIF.gif"
+    ])
+    .expect(page.divUpload.visible)
+    .ok()
+    .expect(page.modalUpload.exists)
+    .ok({ timeout: 50000 })
+    .expect(page.divUpload.child("h4").innerText)
+    .contains("Uploaded 3 out of 3 successfully");
+  await t.takeScreenshot();
+
+  const allPhotosEndCount = await page.allPhotos.count; //Pics count at the end
+  console.log("allPhotosEndCount " + allPhotosEndCount);
+
+  await t.expect(allPhotosEndCount).eql(t.ctx.allPhotosStartCount + 3);
+});
+
+test("Select 1 pic from Photos view", async t => {
+  //Selection bar shows up. It includes AddtoAlbun, Download and Delete buttons
+  await t
+    .hover(page.photoThumb(0))
+    .click(page.photoCheckbox.nth(0)) //Index
+    .expect(page.barPhoto.visible)
+    .ok()
+
+    .expect(page.barPhotoBtnAddtoalbum.visible)
+    .ok()
+    .expect(page.barPhotoBtnDl.visible)
+    .ok()
+    .expect(page.barPhotoBtnDelete.visible)
+    .ok();
+
+  // TODO - Add check on label text ??
+});
+
+test("Select 3 pic from Photos view", async t => {
+  //Selection bar shows up. It includes AddtoAlbun, Download and Delete buttons
+  await t
+    .hover(page.photoThumb(0)) //Only one 'hover' as all checkbox should be visible once the 1st checkbox is checked
+    .click(page.photoCheckbox.nth(0)) //Index
+    .click(page.photoCheckbox.nth(1)) //Index
+    .click(page.photoCheckbox.nth(2)) //Index
+    .expect(page.barPhoto.visible)
+    .ok()
+
+    .expect(page.barPhotoBtnAddtoalbum.visible)
+    .ok()
+    .expect(page.barPhotoBtnDl.visible)
+    .ok()
+    .expect(page.barPhotoBtnDelete.visible)
+    .ok();
+
+  // TODO - Add check on label text ??
```

Comment:
yes plz

Reply:
I'll add label check when we'll add keyword in testcafé, otherwise those checks will have string issues too!